tests/zero-copy/tests/zero-copy.js

```diff
@@ -11,236 +11,281 @@ describe('zero-copy', () => {
 
-    const bar = (
-      await PublicKey.findProgramAddress(
-        [program.provider.wallet.publicKey.toBuffer(), foo.publicKey.toBuffer()],
-        program.programId
+    it('Updates a second zero copy account field', async () => {
+      await program.rpc.updateFooSecond(new BN(55), {
+        accounts: {
+          foo: foo.publicKey,
+          secondAuthority: program.provider.wallet.publicKey,
+        },
+      })
+  
+      const account = await program.account.foo.fetch(foo.publicKey)
+  
+      assert.ok(
+        JSON.stringify(account.authority.toBuffer()) ===
+          JSON.stringify(program.provider.wallet.publicKey.toBuffer())
       )
-    )[0]
-    const barAccount = await program.account.bar.fetch(bar)
-    assert.ok(barAccount.authority.equals(program.provider.wallet.publicKey))
-    assert.ok(barAccount.data.toNumber() === 0)
-  })
-
-  it('Updates an associated zero copy account', async () => {
-    const bar = (
-      await PublicKey.findProgramAddress(
-        [program.provider.wallet.publicKey.toBuffer(), foo.publicKey.toBuffer()],
-        program.programId
+      assert.ok(account.data.toNumber() === 1234)
+      assert.ok(account.secondData.toNumber() === 55)
+      assert.ok(
+        JSON.stringify(account.secondAuthority) ===
+          JSON.stringify([...program.provider.wallet.publicKey.toBuffer()])
       )
-    )[0]
-    await program.rpc.updateBar(new BN(99), {
-      accounts: {
-        bar,
-        authority: program.provider.wallet.publicKey,
-        foo: foo.publicKey,
-      },
     })
-    const barAccount = await program.account.bar.fetch(bar)
-    assert.ok(barAccount.authority.equals(program.provider.wallet.publicKey))
-    assert.ok(barAccount.data.toNumber() === 99)
-    // Check zero_copy CPI
-    await programCpi.rpc.checkCpi(new BN(1337), {
-      accounts: {
-        bar,
-        authority: program.provider.wallet.publicKey,
-        foo: foo.publicKey,
-        zeroCopyProgram: program.programId,
-      },
-    })
-    const barAccountAfterCpi = await program.account.bar.fetch(bar)
-    assert.ok(barAccountAfterCpi.authority.equals(program.provider.wallet.publicKey))
-    assert.ok(barAccountAfterCpi.data.toNumber() === 1337)
-  })
 
-  const eventQ = anchor.web3.Keypair.generate()
-  const size = 1000000 + 8 // Account size in bytes.
-
-  it('Creates a large event queue', async () => {
-    await program.rpc.createLargeAccount({
-      accounts: {
-        eventQ: eventQ.publicKey,
-        rent: anchor.web3.SYSVAR_RENT_PUBKEY,
-      },
-      instructions: [await program.account.eventQ.createInstruction(eventQ, size)],
-      signers: [eventQ],
-    })
-    const account = await program.account.eventQ.fetch(eventQ.publicKey)
-    assert.ok(account.events.length === 25000)
-    account.events.forEach((event) => {
-      assert.ok(event.from.equals(PublicKey.default))
-      assert.ok(event.data.toNumber() === 0)
+    it('Updates a zero copy account field via cpi', async () => {
```

Comment:
@fanatid here's the new test

no load after cpi needed and no cleaning up of references before cpi needed either. it just works like this 😄